### ISSUE TITLE
Enabling multi-threading?

### ISSUE THREAD

**Issue by giordano** [JuliaAstro/SkyCoords.jl#12]
These days I'm pushing a few changes to slightly improve performance of the conversions, and others will come in the next days ;-)

In addition, there is a one-line change that can boost performance: enabling multi-threading
```diff
diff --git a/src/SkyCoords.jl b/src/SkyCoords.jl
index 5aa77d2..a092321 100644
--- a/src/SkyCoords.jl
+++ b/src/SkyCoords.jl
@@ -258,7 +258,7 @@ function convert{T<:AbstractSkyCoords, n, S<:AbstractSkyCoords}(
     ::Type{Array{T,n}}, c::Array{S, n})
     m = Matrix33{_eltype(S)}(rotmat(T, S))
     result = similar(c, T)
-    for i in 1:length(c)
+    Threads.@threads for i in 1:length(c)
         r = m * coords2cart(c[i])
         lon, lat = cart2coords(r)
         result[i] = T(lon, lat)
```

Running the benchmark script at https://github.com/kbarbary/SkyCoords.jl/pull/9#issuecomment-336395556 I get this new plot by using 4 threads:

![bench](https://user-images.githubusercontent.com/765740/31550979-6b55e442-b033-11e7-9cb8-8f7f50cd63e9.png)

With 1 million coordinates `SkyCoords.jl` is still one order of magnitude faster than Astropy :-)  The change was applied on top of current master, so there is still room for some minor improvement.

The problem of multi-threading in Julia is that it's still experimental.  However, when running Julia with just one thread there is zero overhead and no issues.  Thus, the change I propose should be safe enough (doesn't break single-thread systems) and we would only need to warn people that multi-threading may not work.

**Comment by kbarbary** [comment]
Interesting, but I'd rather wait until threading is a bit more stable for this.

Ideally, we'd actually remove the array-wise `convert` method and just use the scalar version and broadcasting. Then threading can just happen at a higher level. We just need to convince the compiler that the rotation matrix can be computed at compile time for any coordinate type pair (it was particularly an issue for `FK5Coords`, where the rotation matrix is different for every parameter `e`). I haven't tried with a recent julia version, but using `@inline` might help.

**Comment by giordano** [comment]
> Interesting, but I'd rather wait until threading is a bit more stable for this.

Yes, that's reasonable.  A few days ago I saw a machine giving garbage results with threading enabled, unfortunately this still happens.

Actually, my next proposed change is going to address the rotation matrix involving FK5 :-)  Stay tuned

**Comment by giordano** [comment]
This issue doesn't apply anymore because the method to convert the arrays has ben removed with #14.  I tried doing a threaded conversion of vectors with this function
```julia
function treads_convert(VT::DataType, c::Vector)
    result = VT(undef, size(c))
    T = eltype(VT)
    Threads.@threads for i in 1:length(c)
        result[i] = convert(T, c[i])
    end
    return result
end
```
but I found a very marginal speedup compared to the serial `convert`, and only for largish arrays.  My blind guess that's because conversion is really fast already and the overhead of the threads scheduler is non-negligible in this case.
